test/http-client-tests/src/main/java/software/amazon/awssdk/http/SdkHttpClientTestSuite.java

```diff
@@ -129,8 +130,15 @@ protected void testForResponseCodeUsingHttps(SdkHttpClient client, int returnCod
     }
 
     private void stubForMockRequest(int returnCode) {
-        stubFor(any(urlPathEqualTo("/")).willReturn(
-                aResponse().withStatus(returnCode).withHeader("Some-Header", "With Value").withBody("hello")));
+        ResponseDefinitionBuilder responseBuilder = aResponse().withStatus(returnCode)
+                                                               .withHeader("Some-Header", "With Value")
+                                                               .withBody("hello");
+
+        if (returnCode >= 300 && returnCode <= 399) {
+            responseBuilder.withHeader("Location", "Some New Location");
+        }
```

Comment:
Do we actually exercise this code path?

Reply:
yes, supportsResponseCode301 and supportsResponseCode302 both trigger it.

Reply:
Without disable leads to test failure:
```
java.lang.NullPointerException: delegate must not be null.

	at software.amazon.awssdk.utils.Validate.paramNotNull(Validate.java:118)
	at software.amazon.awssdk.http.AbortableInputStream.<init>(AbortableInputStream.java:35)
```

Reply:
👍 awesome that you added a base-test for this so that we catch it for other implementations!